src/oic/oic/provider.py

```diff
@@ -1501,6 +1468,39 @@ def verify_redirect_uris(registration_request):
 
         return verified_redirect_uris
 
+    def _verify_sector_identifier(self, request):
+        """
+        Verify `sector_identifier_uri` is reachable and that it contains `redirect uri`s.
+
+        :param request: Provider registration request
+        :return: si_redirects, sector_id
+        :raises: InvalidSectorIdentifier
+        """
+        si_url = request["sector_identifier_uri"]
+        try:
+            res = self.server.http_request(si_url)
+        except ConnectionError as err:
+            logger.error("{}", err)
+            res = None
+
+        if not res:
+            raise InvalidSectorIdentifier("Couldn't open sector_identifier_uri")
+
+        logger.debug("sector_identifier_uri => %s", sanitize(res.text))
+
+        try:
+            si_redirects = json.loads(res.tezt)
+        except ValueError:
+            raise InvalidSectorIdentifier("Error deserializing sector_identifier_uri content")
+
+        if "redirect_uris" in request:
+            logger.debug("redirect_uris: %s". request["redirect_uris"])
+            for uri in request["redirect_uris"]:
```

Comment:
I always find dict comprehension unreadable (personal thing) so I will gladly go with the ignore on this one :)